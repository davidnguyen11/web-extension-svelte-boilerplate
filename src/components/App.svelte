<script lang="ts">
  import { getPalette } from '../lib/get-palette';
  import type { DayPeriod } from '../lib/type/index.type';

  export let date: string;
  export let time: string;
  export let period: DayPeriod;

  const palette = getPalette(period);
</script>

<div style={`background-image: url('images/${period}.png')`} class="root" id="root">
  <div class="root__wrapper">
    <h2 style={`color: ${palette}`} class="root__wrapper__h2">
      {date}
    </h2>
    <h3 style={`color: ${palette}`} class="root__wrapper__h3">
      {time}
    </h3>
  </div>
</div>

<style>
  .root {
    height: 100px;
    width: 240px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-size: cover;
    background-repeat: no-repeat;
  }

  .root__wrapper {
    width: 100%;
    text-align: center;
  }

  .root__wrapper__h2 {
    margin-top: 0;
    margin-bottom: 0;
  }

  .root__wrapper__h3 {
    margin-top: 5px;
    margin-bottom: 0;
  }
</style>
