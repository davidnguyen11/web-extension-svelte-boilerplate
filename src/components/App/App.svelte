<script lang="ts">
  import { getPalette } from '../../lib/get-palette';
  import type { DayPeriod } from '../../type/index.type';
  import AppWrapper from '../AppWrapper/AppWrapper.svelte';
  import DateTime from '../DateTime/DateTime.svelte';

  export let date: string;
  export let time: string;
  export let period: DayPeriod;

  const palette = getPalette(period);
</script>

<AppWrapper {period}>
  <div class="root__wrapper">
    <DateTime {date} {time} {palette} />
  </div>
</AppWrapper>

<style>
  .root__wrapper {
    width: 100%;
    text-align: center;
  }
</style>
