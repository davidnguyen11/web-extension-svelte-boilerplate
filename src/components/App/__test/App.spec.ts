import App from '../App.svelte';
import { render } from '@testing-library/svelte';

describe('testing getDisplayDateTime', () => {
  it('should display datetime', async () => {
    const AppRendered = render(App, {
      date: 'Fri May 14',
      time: '16:44',
      period: 'afternoon',
    });

    expect(AppRendered.container).toMatchSnapshot();
  });
});
