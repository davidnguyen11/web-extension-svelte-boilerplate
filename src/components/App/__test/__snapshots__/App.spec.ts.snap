// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing getDisplayDateTime should display datetime 1`] = `
<body>
  <div>
    <div
      class="root svelte-1tq4r33"
      role="main"
      style="background-image: url(images/afternoon.png)"
    >
      <div
        class="root__wrapper svelte-hw1lzg"
      >
        <h2
          class="datetime__h2 svelte-1pg3641"
          style="color: #000000"
        >
          Fri May 14
        </h2>
         
        <h3
          class="datetime__h3 svelte-1pg3641"
          style="color: #000000"
        >
          16:44
        </h3>
      </div>
    </div>
  </div>
</body>
`;
